pkg/test/clusterdeployment/clusterdeployment.go

```diff
@@ -125,6 +125,12 @@ func WithClusterPoolReference(namespace, poolName, claimName string) Option {
 	}
 }
 
+func PreserveOnDelete() Option {
+	return func(clusterDeployment *hivev1.ClusterDeployment) {
+		clusterDeployment.Spec.PreserveOnDelete = true
+	}
+}
+
```

Comment:
Looks like you didn't end up using this? I took it out and everything still works. (No harm keeping it "for the future", but YAGNI.)

Reply:
Right again, although tempted to keep this one so it's there. If we converted all our tests to use this framework, it would get used.

Reply:
Right, the YAGNI principle says we add it in if/when that ever happens. But as you wish, not a show-stopper either way.